iothub/device/src/Authentication/IotHubConnectionCredentials.cs

```diff
@@ -291,23 +299,13 @@ private void SetAuthenticationModel()
         private void Validate()
         {
             // IoT Hub host name
-            if (IotHubHostName.IsNullOrWhiteSpace())
-            {
-                throw new FormatException("IoT hub service hostname to connect to must be specified.");
-            }
+            Argument.AssertNotNullOrWhiteSpace(IotHubHostName, nameof(IotHubHostName));
```

Comment:
Updated the FormatException to ArgumentException/ArgumentNullException. We encounter this even in the non-connection string flow, i.e. through an IAuthenticationMethod flow, so it made sense to throw the generic argument exceptions.